dotCMS/src/main/java/com/dotcms/webdav/FsMemoryLockManager.java

```diff
@@ -0,0 +1,141 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package com.dotcms.webdav;
+
+import com.dotcms.repackage.com.bradmcevoy.http.*;
+import com.dotcms.repackage.com.bradmcevoy.http.exceptions.NotAuthorizedException;
+import com.dotcms.repackage.com.ettrema.http.fs.LockManager;
+import com.dotmarketing.util.Logger;
+
+import java.io.File;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+
+/**
+ *
+ */
+public class FsMemoryLockManager implements LockManager {
+
+
+    /**
+     * maps current locks by the file associated with the resource
+     */
+    Map<File, CurrentLock> locksByFile;
+    Map<String, CurrentLock> locksByToken;
+
+    public FsMemoryLockManager() {
+        locksByFile = new HashMap<File, CurrentLock>();
+        locksByToken = new HashMap<String, CurrentLock>();
+    }
+
+    @Override
+    public synchronized LockResult lock( LockTimeout timeout, LockInfo lockInfo, LockableResource r ) {
+        FsResource resource = (FsResource) r;
+        LockToken currentLock = currentLock( resource );
+        if( currentLock != null ) {
+            return LockResult.failed( LockResult.FailureReason.ALREADY_LOCKED );
+        }
+
+        LockToken newToken = new LockToken( UUID.randomUUID().toString(), lockInfo, timeout );
+        CurrentLock newLock = new CurrentLock( resource.getFile(), newToken, lockInfo.lockedByUser );
+        locksByFile.put( resource.getFile(), newLock );
+        locksByToken.put( newToken.tokenId, newLock );
+        return LockResult.success( newToken );
+    }
+
+    @Override
+    public synchronized LockResult refresh( String tokenId, LockableResource resource ) {
+        CurrentLock curLock = locksByToken.get( tokenId );
+        if( curLock == null ) {
+          Logger.info(FileSystemResourceFactory.class, "can't refresh because no lock");
+            return LockResult.failed( LockResult.FailureReason.PRECONDITION_FAILED );
+        } else {
+            curLock.token.setFrom( new Date() );

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use the Java 8 Date and Time API instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2143)